ceresdb/client.go

```diff
@@ -4,67 +4,19 @@ package ceresdb
 
 import (
 	"context"
-	"errors"
-
-	"github.com/CeresDB/ceresdbproto/go/ceresdbproto"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials/insecure"
-)
-
-const (
-	codeOk = 200
+	"github.com/CeresDB/ceresdb-client-go/types"
 )
 
-type Client struct {
-	inner ceresdbproto.StorageServiceClient
-	conn  *grpc.ClientConn
-}
-
-func NewClient(addr string) (*Client, error) {
-	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(insecure.NewCredentials()),
-		grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(1024*1024*1024)))
-	if err != nil {
-		return nil, err
-	}
-
-	c := ceresdbproto.NewStorageServiceClient(conn)
-
-	return &Client{
-		inner: c,
-		conn:  conn,
-	}, nil
-}
-
-func (c *Client) Write(ctx context.Context, points []Point) (int, error) {
-	req := writeRequest{
-		points: points,
-	}
-
-	resp, err := c.inner.Write(ctx, req.toPb())
-	if err != nil {
-		return 0, err
-	}
-
-	if codeOk == resp.Header.Code {
-		return int(resp.Success), nil
-	}
-
-	return 0, errors.New(resp.Header.GetError())
-}
-
-func (c *Client) Close() error {
-	return c.conn.Close()
+type CeresDBClient interface {
+	Query(context.Context, *types.QueryRequest) (*types.QueryResponse, error)
+	Write(context.Context, []*types.Row) (*types.WriteResponse, error)
```

Comment:
How about define a `WriteRequest` type?

Reply:
Temporarily consistent with java SDK.
And if the SDK redefines the api later, use WriteQuest uniformly, and then modify it.